tot/data_processing/scaler.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+
+import numpy as np
+import pandas as pd
+
+
+@dataclass
+class Scaler:
+    transformer: object
+
+    def __post_init__(self):
+        is_transformer_valid = (
+            callable(getattr(self.transformer, "fit", None))
+            and callable(getattr(self.transformer, "transform", None))
+            and callable(getattr(self.transformer, "inverse_transform", None))
+        )
+        if not is_transformer_valid:
+            raise ValueError(
+                "Transformer provided to the Scaler must implement fit, transform and inverse_transform " "methods"
+            )
+
+    def _transform(self, df: pd.DataFrame, fit):
+        values = []
+        for df_name, df_i in df.groupby("ID"):
+            aux = df_i.copy(deep=True)
+            values.append(aux["y"])
+
+        values = np.transpose(values)
+        if fit:
+            self.transformer.fit(values)
+        values_transformed = np.transpose(self.transformer.transform(values))
+
+        result = pd.DataFrame()
+        for (_, df_i), values_row in zip(df.groupby("ID"), values_transformed):
+            group = df_i.copy(deep=True)
+            group.loc[:, "y"] = values_row
+            result = pd.concat((result, group))
+
+        return result
+
+    def _inverse_transform(self, df: pd.DataFrame):
+        values_pred = []
+        values_true = []
+        for df_name, df_i in df.groupby("ID"):
```

Comment:
When adjusting _transform() to directly pass a df, we can adjust it here as well